packages/notifi-frontend-client/lib/client/NotifiFrontendClient.ts

```diff
@@ -426,23 +430,27 @@ export type Uint8SignMessageFunction = (
 ) => Promise<Uint8Array>;
 export type AptosSignMessageFunction = (
   message: string,
-  nonce: number,
-) => Promise<string>;
-type hexString = `0x${string}`;
+  nonce: string,
+) => Promise<{
+  signatureHex: HexString;
+  signedMessage: string;
+}>;
 
 export type AcalaSignMessageFunction = (
   acalaAddress: string,
   message: string,
-) => Promise<hexString>;
+) => Promise<HexString>;
 
 export type OidcCredentials = {
   oidcProvider: Types.OidcProvider;
   jwt: string;
 };
+
 export type OidcSignInFunction = () => Promise<OidcCredentials>;
 
-export type AuthenticateResult = Signature | OidcCredentials;
-type Signature = string;
+export type AuthenticateResult =
+  | { signature: string; signedMessage: string }
```

Comment:
I think changing the interface here will break the login method becuase below in line746,  772 - 775, 807 - 810.